cranelift-codegen/meta/src/shared/legalize.rs

```diff
@@ -494,16 +479,6 @@ pub fn define(insts: &InstructionGroup, immediates: &OperandKinds) -> TransformG
         ],
     );
 
-    expand.legalize(
-        def!((a, c) = iadd_carry(x, y, c_in)),
-        vec![
-            def!((a1, c1) = iadd_cout(x, y)),
-            def!(c_int = bint(c_in)),
-            def!((a, c2) = iadd_cout(a1, c_int)),
-            def!(c = bor(c1, c2)),
-        ],
```

Comment:
Those legaizations are useful for platforms which dont yet support them natively. It wont hurt for x86 to keep them, as legalization is only performed when there is no encoding for an instruction.

Reply:
You're right. But this also implies that we can't just use the iflags as the type variable for the carry flags, which makes it impossible to properly implement add-with-carry on i686. @ryzokuken discussed this with Dan, who suggested removing the legalizations, since they were useful only for RISCV, which isn't quite properly implemented anyway. We'll try to find a better solution, like introducing new instructions maybe (iadd_ifcout/iadd_ifcin/iadd_ifcarry), that'll be involved on x86 legalizations for iadd_cout/cin/carry.

Reply:
+1 @bnjbvr. See https://github.com/CraneStation/cranelift/pull/961/files#diff-8d295395253ef7b2fd5b41fdd6386e73R1 and the commit messages for more.